apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient5-plugin/src/test/java/co/elastic/apm/agent/httpclient/v5/ApacheHttpAsyncClientInstrumentationTest.java

```diff
@@ -68,24 +66,9 @@ protected void performGet(String path) throws Exception {
             .build();
         HttpClientContext httpClientContext = HttpClientContext.create();
         httpClientContext.setRequestConfig(requestConfig);
-        client.execute(req, httpClientContext, new FutureCallback<SimpleHttpResponse>() {
-            @Override
-            public void completed(SimpleHttpResponse simpleHttpResponse) {
-                responseFuture.complete(simpleHttpResponse);
-            }
-
-            @Override
-            public void failed(Exception e) {
-                responseFuture.completeExceptionally(e);
-            }
 
-            @Override
-            public void cancelled() {
-                responseFuture.cancel(true);
-            }
-        });
-
-        responseFuture.get();
+        // using the returned future to wait on request completion, using an explicit callback is covered by performPost
+        client.execute(req, httpClientContext, null).get();
```

Comment:
[for reviewer] this emulates what is being done in the ES driver 9.x, using a `null` callback and then waiting on the returned future. I initially thought that the issue was due to wrapping `null` here, but the issue was in fact about recycling objects. I've left it as it allows to ensure that we have unit test with `null` callback.